modin/engines/base/io/io.py

```diff
@@ -11,51 +11,110 @@
 # ANY KIND, either express or implied. See the License for the specific language
 # governing permissions and limitations under the License.
 
+"""
+Module houses `BaseIO` class.
+
+`BaseIO` is base class for IO classes, that stores IO functions.
+"""
+
 import pandas
+from pandas.util._decorators import doc
 from collections import OrderedDict
 from modin.error_message import ErrorMessage
 from modin.backends.base.query_compiler import BaseQueryCompiler
 from typing import Optional
 
+# TODO (amyskov): replace `For parameters description please refer to Pandas API.` statement with
+# @_inherit_docstrings decorator when #2969 will be merged.
+_doc_default_io_method = """
+{summary} using Pandas.
+
+For parameters description please refer to Pandas API.
+
+Returns
+-------
+{returns}
+"""
+
+_doc_returns_qc = """BaseQueryCompiler
+    QueryCompiler with read data."""
+
+_doc_returns_qc_or_parser = """BaseQueryCompiler or TextParser
+    QueryCompiler or TextParser with read data."""
+
 
 class BaseIO(object):
+    """Class for basic utils and default implementation of IO functions."""
+
     query_compiler_cls: BaseQueryCompiler = None
     frame_cls = None
 
     @classmethod
     def from_non_pandas(cls, *args, **kwargs):
+        """
+        Improve non-pandas object to an advanced Modin query compiler.
```

Comment:
I would prefer to have more clear docstrings for similar functions rather than it is now. An example:
```
Create a <Modin object> from an <object>.
```

Reply:
Rewritten as ``Create a Modin `query_compiler` from a non-pandas `object`.``, for highlighting `` ` `` symbols are used.